### PR TITLE
Sort translations in .desktop files

### PR DESCRIPTION
**Pull request by ThomasVie**
For me it is annoying to search for a locale if the .desktop files are not sorted. This patch will do it.
Additionally fixed some typos.

### CODE REVIEW

File: cmake/modules/LXQtTranslateDesktop.cmake [hunk 1]
```diff
@@ -72,7 +72,7 @@ function(lxqt_translate_desktop _RESULT)
         if (_translations)
             add_custom_command(OUTPUT ${_outFile}
                 COMMAND grep -v "'#TRANSLATIONS_DIR='" ${_inFile} > ${_outFile}
-                COMMAND grep -h ${_pattern} ${_translations} >> ${_outFile}
+                COMMAND grep -h ${_pattern} ${_translations} | sort >> ${_outFile}
```
Comment (palinek) [inline cmake/modules/LXQtTranslateDesktop.cmake]: It's a matter of taste, but IMO it's better to have all keys for one language together...

**Comment by ThomasVie** [comment]
So, well, do you know a way to sort the file names ${_translations}?

**Comment by palinek** [comment]
Well, we have 2 options (as I see it):
- sort the list of files `_translations`:
```diff
--- a/cmake/modules/LXQtTranslateDesktop.cmake
+++ b/cmake/modules/LXQtTranslateDesktop.cmake
@@ -67,6 +67,7 @@ function(lxqt_translate_desktop _RESULT)
             ${_translationDir}/${_fileName}_*${_fileExt}
             ${_translationDir}/local/${_fileName}_*${_fileExt}
         )
+        list(SORT _translations)
 
         set(_pattern "'\\[.*]\\s*='")
         if (_translations)
```
, but there is a problem that files are searched in 2 directories. @luis-pereira why is also the `local` subdirectory searched (is it used anywhere?)?

- or instruct `sort` to sort based on key in [ ]:
```diff
--- a/cmake/modules/LXQtTranslateDesktop.cmake
+++ b/cmake/modules/LXQtTranslateDesktop.cmake
@@ -72,7 +72,7 @@ function(lxqt_translate_desktop _RESULT)
         if (_translations)
             add_custom_command(OUTPUT ${_outFile}
                 COMMAND grep -v "'#TRANSLATIONS_DIR='" ${_inFile} > ${_outFile}
-                COMMAND grep -h ${_pattern} ${_translations} >> ${_outFile}
+                COMMAND grep -h ${_pattern} ${_translations} | sort -k2 -t[ >> ${_outFile}
                 COMMENT "Generating ${_fileName}${_fileExt}"
             )
         else()
```
, but this has the problem that the key for sorting are characters from [ to the end of line (so for each translation the keys Name, Comment ... are not in the same order)

**Comment by luis-pereira** [comment]
@ThomasVie The place to localize is lxqt-component_locale.desktop. The lxqt-component.desktop is for machine consumption.
@palinek It came from razor. It's an artifact from that time. Soon to be removed. I'm on updating the build system.

**Comment by ThomasVie** [comment]
> The lxqt-component.desktop is for machine consumption.
Yes, I know. But it would be nice to have a reproducible build - so that you and I
produce the same product and all files were bytewise identical.
To sort would be one possibility to achieve it.

**Pull request closed without merging**